### PR TITLE
Windows.Devices.Lights.Lamp

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): #2017

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

No support for `Windows.Devices.Lights.Lamp`

## What is the new behavior?

Functional support for `Windows.Devices.Lights.Lamp` on Android and iOS

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_Devices/LampTests.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+using Uno.UI.Samples.UITests.Helpers;
+using Windows.Devices.Lights;
+using Windows.UI.Core;
+using Windows.UI.Xaml.Controls;
+
+namespace UITests.Shared.Windows_Devices
+{
+	[SampleControlInfo(
+		"Windows.Devices",
+		"Lamp",
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/Windows_Devices/LampTests.xaml.cs]: By convention we use the class name here.
(by the way it won't be required anymore once my PR #1726 is merged!)

Comment (MartinZikmund) [reply src/SamplesApp/UITests.Shared/Windows_Devices/LampTests.xaml.cs]: In the `Windows.Devices` category, all the samples have their name the same as their class name. In fact, I have done it this way for the majority of my PRs so far :-D .

Comment (carldebilly) [reply src/SamplesApp/UITests.Shared/Windows_Devices/LampTests.xaml.cs]: I mean `Lamp` vs `LampTests`.

Comment (MartinZikmund) [reply src/SamplesApp/UITests.Shared/Windows_Devices/LampTests.xaml.cs]: I meant that I have done it in similar fashion on the other sensors, for example here https://github.com/unoplatform/uno/blob/0430120103d6552443e2db7c5e0aaebcce059b2e/src/SamplesApp/UITests.Shared/Windows_Devices/BarometerTests.xaml.cs#L23 and here https://github.com/unoplatform/uno/blob/0430120103d6552443e2db7c5e0aaebcce059b2e/src/SamplesApp/UITests.Shared/Windows_Devices/GeolocatorTests.xaml.cs#L16

File: doc/articles/features/windows-device-lights.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## `Lamp`
+
+The following APIs are implemented on Android and iOS:
+
+* `GetDefaultAsync`
+* `IsEnabled`
+* `BrightnessLevel`
+
+WASM does not offer a API to control the device flashlight, yet.
```
Comment (jeromelaban) [inline doc/articles/features/windows-device-lights.md]:
```suggestion
WASM does not offer an API to control the device flashlight, yet.
```

**Comment by jeromelaban** [comment]
Thanks Martin!

File: src/Uno.UWP/Devices/Lights/Lamp.Android.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+namespace Windows.Devices.Lights
+{
+	public partial class Lamp
+	{
+		private readonly object _lock = new object();
+
+		private readonly string _defaultCameraId;
+		private CameraManager _cameraManager;
+
+#pragma warning disable CS0618
```
Comment (davidjohnoliver) [inline src/Uno.UWP/Devices/Lights/Lamp.Android.cs]: Add a comment here explaining why the deprecated API is used

File: src/Uno.UWP/Devices/Lights/Lamp.Android.cs [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+						_camera.StopPreview();
+					}
+#pragma warning restore CS0618
+				}
+			}
+		}
+
+
+		public void Dispose()
+		{
```
Comment (davidjohnoliver) [inline src/Uno.UWP/Devices/Lights/Lamp.Android.cs]: should all probably be inside the lock

Comment (MartinZikmund) [reply src/Uno.UWP/Devices/Lights/Lamp.Android.cs]: Done 👌

File: src/Uno.UWP/Devices/Lights/Lamp.Android.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UWP/Devices/Lights/Lamp.Android.cs]: Done 👍

**Pull request merged**